### PR TITLE
Migrate WDPC to roles

### PR DESCRIPTION
**Pull request by danieljames-dj**
WDPC is the only team that doesn't exist anymore. We haven't created user_group as well for this team. So this migration is also creating a user_group for WDPC.

### CODE REVIEW

File: db/schema.rb
Comment (gregorbg) [inline db/schema.rb:1]: This has unrelated changes

File: db/migrate/20240503144557_migrate_wdpc_to_groups.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+class MigrateWdpcToGroups < ActiveRecord::Migration[7.1]
+  include RoleMigrationHelper
+
+  def change
+    team = Team.find_by_friendly_id('wdpc')
+    # email was removed in https://github.com/thewca/worldcubeassociation.org/pull/6065, but I don't
+    # think there was any reason to remove it. Hence adding it back.
+    metadata = GroupsMetadataTeamsCommittees.create!(email: 'dataprotection@worldcubeassociation.org', friendly_id: team.friendly_id)
+    UserGroup.create!(name: team.name, group_type: UserGroup.group_types[:teams_committees], is_active: false, is_hidden: false, metadata: metadata)
+    migrate_team_members_to_group(Team.c_find_by_friendly_id!('wdpc'), GroupsMetadataTeamsCommittees.find_by(friendly_id: 'wdpc').user_group)
```
Comment (gregorbg) [inline db/migrate/20240503144557_migrate_wdpc_to_groups.rb:12]: You have `Team.find_by_friendly` already stored in a variable above. Just reuse that

File: db/migrate/20240503144557_migrate_wdpc_to_groups.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+class MigrateWdpcToGroups < ActiveRecord::Migration[7.1]
+  include RoleMigrationHelper
+
+  def change
+    team = Team.find_by_friendly_id('wdpc')
+    # email was removed in https://github.com/thewca/worldcubeassociation.org/pull/6065, but I don't
+    # think there was any reason to remove it. Hence adding it back.
+    metadata = GroupsMetadataTeamsCommittees.create!(email: 'dataprotection@worldcubeassociation.org', friendly_id: team.friendly_id)
+    UserGroup.create!(name: team.name, group_type: UserGroup.group_types[:teams_committees], is_active: false, is_hidden: false, metadata: metadata)
```
Comment (gregorbg) [inline db/migrate/20240503144557_migrate_wdpc_to_groups.rb:11]: This `create` call returns the Rails model that you can store into a variable and use directly below

File: db/seeds/user_groups.seeds.rb [hunk 3]
```diff
@@ -233,14 +233,20 @@
...
     group_type: :teams_committees,
     is_active: true,
     is_hidden: true,
     metadata: GroupsMetadataTeamsCommittees.find_by!(friendly_id: 'wct_china'),
   )
+  UserGroup.create!(
+    name: 'WCA Data Protection Committee',
+    group_type: :teams_committees,
+    is_active: false,
+    is_hidden: false,
```
Comment (gregorbg) [inline db/seeds/user_groups.seeds.rb:247]: uh oh

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: db/migrate/20240503144557_migrate_wdpc_to_groups.rb (hunk 2 above)
Comment (danieljames-dj) [reply db/migrate/20240503144557_migrate_wdpc_to_groups.rb]: Oh ya, thanks. Done.

File: db/migrate/20240503144557_migrate_wdpc_to_groups.rb (hunk 1 above)
Comment (danieljames-dj) [reply db/migrate/20240503144557_migrate_wdpc_to_groups.rb]: Oops, done.

File: db/schema.rb
Comment (danieljames-dj) [reply db/schema.rb:1]: My mistake, removed 😅

File: db/seeds/user_groups.seeds.rb (hunk 3 above)
Comment (danieljames-dj) [reply db/seeds/user_groups.seeds.rb:247]: It was intentional actually. WDPC 'team' is hidden because it is no longer active but there was no way in teams table to set active status. Here, `is_active` solves that. Hence, if we decide to show past roles of a user in a public page, we have to show their WDPC role (if exists) as well and for that the group should not be hidden.

**Review by gregorbg — APPROVED** [review]
